File: game-server/src/com/aionemu/gameserver/network/aion/capture/NoOpServerPacketCaptureObserver.java
package com.aionemu.gameserver.network.aion.capture;

import java.nio.ByteBuffer;

import com.aionemu.gameserver.network.aion.AionConnection;
import com.aionemu.gameserver.network.aion.AionServerPacket;

/**
 * Disabled default packet capture observer.
 */
public enum NoOpServerPacketCaptureObserver implements ServerPacketCaptureObserver {
	INSTANCE;

	@Override
	public boolean isEnabled() {
		return false;
	}

	@Override
	public void onPacketSerialized(AionConnection con, AionServerPacket packet, ByteBuffer clearFrame) {
	}
}

File: dotnetConversion/src/Aion.GameServer/Network/Aion/Capture/NoOpServerPacketCaptureObserver.cs
using Aion.Commons.Nio;
using Aion.GameServer.Network.Aion;

namespace Aion.GameServer.Network.Aion.Capture;

/// <summary>
/// Java parity: network/aion/capture/NoOpServerPacketCaptureObserver (enum singleton INSTANCE) ->
/// C# singleton class (C# enums can't implement interface methods). No-op observer.
/// </summary>
public sealed class NoOpServerPacketCaptureObserver : ServerPacketCaptureObserver
{
    public static readonly NoOpServerPacketCaptureObserver INSTANCE = new NoOpServerPacketCaptureObserver();

    private NoOpServerPacketCaptureObserver() { }

    public bool IsEnabled() => false;

    public void OnPacketSerialized(AionConnection con, AionServerPacket packet, ByteBuffer clearFrame) { }
}
